pkg/azuredisk/azuredisk.go

```diff
@@ -401,18 +407,36 @@ func getSourceVolumeId(snapshot *compute.Snapshot) string {
 	return ""
 }
 
-func getValidCreationData(subscriptionID, resourceGroup, sourceResourceID string) (compute.CreationData, error) {
+func getValidCreationData(subscriptionID, resourceGroup, sourceResourceID, sourceType string) (compute.CreationData, error) {
 	if sourceResourceID == "" {
 		return compute.CreationData{
 			CreateOption: compute.Empty,
 		}, nil
 	}
-	if match := diskSnapshotPathRE.FindString(sourceResourceID); match == "" {
-		sourceResourceID = fmt.Sprintf(diskSnapshotPath, subscriptionID, resourceGroup, sourceResourceID)
+
+	switch sourceType {
+	case sourceSnapshot:
```

Comment:
make sourceType as case insensitive?

Reply:
Since sourceType is set by ourselves, there are just two values for it. We do not need to care about case sensitive